utensor_cgen/transformer/ns_transformer.py

```diff
@@ -13,7 +13,175 @@
 
 from .base import Transformer
 
-__all__ = ["DropoutTransformer", "BatchNormTransformer", "InlineTransformer"]
+__all__ = ["DropoutTransformer", "BatchNormTransformer", "InlineTransformer", "TensorLifeProbe"]
+class TensorLifeProbe(Transformer):
+  METHOD_NAME = 'tensorlife'
+  KWARGS_NAMESCOPE = '_utensor_utlife'
+
+
+  def transform(self, ugraph):
+    use_def_table = self._create_resource_table(ugraph)
+    unit_size = 4
+    allocate_success = False
+    buffer_size = 50 #1k bytes
+    while not allocate_success:
+      allocate_table = dict()
+      for node_name in ugraph.topo_order:
+
+        success = False
+        in_t_infos = ugraph.ops_info[node_name].input_tensors  # may be input tensor doest not need this
+        for in_o in in_t_infos:
+          for i in range(0, buffer_size, unit_size):
+            tensor_size = self._getSize(in_o.shape)
+            if self._check(ugraph, allocate_table, use_def_table, in_o, i, i + tensor_size):
+              self._create_allocate_table(allocate_table, use_def_table, in_o, i, i + tensor_size)
+              break
+            else:
+              success = False 
```

Comment:
Looks like after this loop, `success` is always `False`.
Is this what you want?

Also, thought not very well-known, there is `else` block for `for` loop in python.